x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/function/scalar/whitelist/InternalQlScriptUtils.java

```diff
@@ -58,6 +59,10 @@ public static Number nullSafeCastNumeric(Number number, String typeName) {
         return number == null || Double.isNaN(number.doubleValue()) ? null : (Number) convert(number, fromTypeName(typeName));
     }
 
+    public static Number nullSafeCastToUnsignedLong(Number number) {
```

Comment:
I think this new method creates a bwc issue in mixed node environments. Maybe consider bumping the `minCompatibleShardNode` version in `org/elasticsearch/xpack/sql/execution/search/Querier.java:147` and cover it in `SqlCompatIT`.

Reply:
Changed the `minCompatibleShardNode`. 
The test with bwcVersion nodes would fail indeed, but due to missing UL SQL support, not just missing script, while script support current version nodes is validated in other tests, so not sure how to add a meaningful test specifically for this.